File: ObjC/CustomTemplatesExample/CustomTemplatesExample/SliderTemplate/SlideViewController.m
//
//  PageViewController.m
//  CustomTemplatesExample
//
//  Created by Nikola Zagorchev on 27.09.20.
//  Copyright © 2020 Nikola Zagorchev. All rights reserved.
//

#import "SlideViewController.h"
#import "LPFileManager.h"

@interface SlideViewController ()

@end

@implementation SlideViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [[self titleLabel] setText:self.titleText];
    [[self titleLabel] setTextColor:self.titleColor];

    if (self.imageName){
        // Try to download the image
        BOOL willDownload = [LPFileManager maybeDownloadFile:self.imageName defaultValue:nil onComplete:^{
            [self setImageToView];
        }];
        
        // If the image is already downloaded it will not trigger the onComplete callback
        if (!willDownload) {
            [self setImageToView];
        }
    }
}

- (void)setImageToView {
    UIImage *img = [self getImageFile:self.imageName];
    [self.imageView setImage:img];
}

/**
 * Get the image from the file system
 */
- (UIImage*)getImageFile:(NSString*)fileName {
    NSString *fileValue = [LPFileManager fileValue:fileName withDefaultValue:fileName];
    if (fileValue) {
        UIImage *img = [UIImage imageWithContentsOfFile:fileValue];
        return img;
    }
    return nil;
}
@end

File: Swift/CustomTemplatesExample/CustomTemplatesExample/Slider/SlideViewController.swift
//
//  SlideViewController.swift
//  CustomTemplatesExample
//
//  Created by Nikola Zagorchev on 6.10.20.
//  Copyright © 2020 Nikola Zagorchev. All rights reserved.
//

import UIKit
import Leanplum

class SlideViewController: UIViewController {
    var imageName:String?
    var titleText:String?
    var titleColor:UIColor?
    var pageIndex:Int?
    
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var imageView: UIImageView!
    
    override func viewDidLoad() {
        
        if let text = titleText {
            self.titleLabel.text = text
        }
        
        if let color = titleColor {
            self.titleLabel.textColor = color
        }
        
        if let fileName = imageName {
            // try to download the image
            // if the image is already downloaded it will not trigger the onComplete callback
            let willDownload = LPFileManager.maybeDownloadFile(fileName, defaultValue: "", onComplete: {
                self.setImage(img: self.getImageFile(fileName))
            })
            
            if !willDownload {
                setImage(img: getImageFile(fileName))
            }
        }
    }
    
    func setImage(img:UIImage?) {
        if let image = img {
            self.imageView.image = image
        }
    }
    
    func getImageFile(_ fileName: String) -> UIImage? {
        let fl = LPFileManager.fileValue(fileName, withDefaultValue: fileName)
        if let file = fl {
            let img = UIImage.init(contentsOfFile: file)
            return img
        }
        return nil
    }
}
